proposals/init.md

```diff
@@ -0,0 +1,490 @@
+true for a property. 
+
+If we want to make the removal of `init` from a property a compatible change
+then it will force our hand on the modreq vs. attributes decision below. If 
+one the other hand this is seen as a non-interesting scenario then this will 
+make the modreq vs. attribute decision less impactful.
+
+**Resolution**
+This scenario is not seen as compelling by LDM.
+
+### Modreqs vs. attributes
+The emit strategy for `init` property accessors must choose between using 
+attributes or modreqs when emitting during metadata. These have different 
+trade offs that need to be considered.
+
+Annotating a property set accessor with a modreq declaration means CLI compliant
+compilers will ignore the accessor unless it understands the modreq. That means
+only compilers aware of `init` will read the member. Compilers unaware of 
+`init` will ignore the `set` member and hence will not accidentally treat the
+property as read / write. 
+
+The downside of modreq is `init` becomes a part of the binary signature of 
+the `set` accessor. Adding or removing `init` will break binary compatbility 
+of the application.
+
+Using attributes to annotate the `set` accessor means that only compilers which
+understand the attribute will know to limit access to it. A compiler unaware 
+of `init` will see it as a simple read / write property and allow access.
+
+This would seemingly mean this decision is a choice between extra safety at 
+the expense of binary compatibility. Digging in a bit the extra safety is not
+exactly what it seems. It will not for instance protect against the following
+circumstances:
+
+1. Reflection over `public` members
+1. The use of `dynamic` 
+1. Compilers that don't recognize modreqs
+
+It should also be considered that when we complete the IL verification rules 
+for .NET 5, `init` will be one of those rules. that means extra enforcement 
+will be gained from simply verifying compilers emitting verifiable IL.
+
+The primary languages for .NET (C#, F# and VB) will all be updated to 
+recognize these `init` accessors. Hence the only realistic scenario here is 
+when a C# 9 compiler emits `init` properties and they are seen by a C# 8 
+compiler. That is the trade off to consider and weigh against binary 
+compatibility.
+
+Note: this discussion applies to the accessor members only, not to fields. There
+is no value to be gained by using a modreq on a field. The `init` feature for 
+fields is a relaxation of an existing rule. All existing compilers already 
+support `readonly` and hence an attribute serves fine as a way to alert them
+that write access can be extended in certain circumstances.
+
+**Resolution**
+The feature will use a modreq to encode the property `init` setter. The
+compelling factors were (in no particular order):
+
+1. Desire to discourage older compilers from violating `init` semantics
+1. Desire to make adding or removing `init` in a `virtual` declaratio or 
```

Comment:
```suggestion
* Desire to discourage older compilers from violating `init` semantics
* Desire to make adding or removing `init` in a `virtual` declaratio or 
```

Reply:
Have a typo on `declaratio` I need to fix here.